File: Arrays/two_sum.py
# Link: https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        buffer_dictionary = {}
        list_len = len(nums)
        for i in range(list_len):
            if nums[i] in buffer_dictionary:
                return [buffer_dictionary[nums[i]], i]
            else:
                buffer_dictionary[target- nums[i]] = i
            

File: Algor_Questions/Arrays/two_sum.js
var twoSum = function(nums, target) {
    let myArr = new Map();
    for(var idx in nums){
        if(myArr.has(target - nums[idx])){
            return [idx,myArr.get(target - nums[idx])];
        }else{
            myArr.set(nums[idx],idx);
        }
    }
    
};
